crates/storage/codecs/src/alloy/genesis_account.rs

```diff
@@ -0,0 +1,67 @@
+use crate::Compact;
+use alloy_genesis::GenesisAccount as AlloyGenesisAccount;
+use alloy_primitives::{Bytes, B256, U256};
+use reth_codecs_derive::main_codec;
+
+/// GenesisAccount acts as bridge which simplifies Compact implementation for AlloyGenesisAccount.
+///
+/// Notice: Make sure this struct is 1:1 with `alloy_genesis::GenesisAccount`
+#[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+struct GenesisAccount {
+    /// The nonce of the account at genesis.
+    nonce: Option<u64>,
+    /// The balance of the account at genesis.
+    balance: U256,
+    /// The account's bytecode at genesis.
+    code: Option<Bytes>,
+    /// The account's storage at genesis.
+    storage: Option<StorageEntries>,
+    /// The account's private key. Should only be used for testing.
+    private_key: Option<B256>,
+}
+
+#[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+struct StorageEntries {
+    entries: Vec<StorageEntry>,
+}
+
+#[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+struct StorageEntry {
+    key: B256,
+    value: B256,
+}
+
+impl Compact for AlloyGenesisAccount {
```

Comment:
why do we need this? when are we writing the genesis account to database?

Reply:
We don't, but `reth_etl::Collector` requires `V` to impl `Compact`

On early implementations, i was trying to leverage its `serde` with some other serializing lib, but it was failing on deserializing, and was only succeeding with `serde_json` which is too slow.